operator/internal/handlers/internal/tlsprofile/tlsprofile.go

```diff
@@ -0,0 +1,71 @@
+package tlsprofile
+
+import (
+	"context"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	projectconfig_types "github.com/grafana/loki/operator/apis/config/v1"
+	"github.com/grafana/loki/operator/internal/external/k8s"
+	openshiftv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/crypto"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetSecurityProfileInfo gets the tls profile info to apply.
+func GetSecurityProfileInfo(ctx context.Context, k k8s.Client, tlsProfileType projectconfig_types.TLSProfileType) (projectconfig_types.TLSProfileSpec, error) {
+	var tlsProfile openshiftv1.TLSSecurityProfile
+
+	if tlsProfileType != "" {
+		tlsProfile = openshiftv1.TLSSecurityProfile{
+			Type: openshiftv1.TLSProfileType(tlsProfileType),
+		}
+	} else {
+		tlsProfile = openshiftv1.TLSSecurityProfile{
+			Type: openshiftv1.TLSProfileIntermediateType,
+		}
+
+		var apiServer openshiftv1.APIServer
+		if err := k.Get(ctx, client.ObjectKey{Name: "cluster"}, &apiServer); err != nil {
```

Comment:
Let's make the name a constant to make this visible that we support a single apiserver.